sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -626,15 +627,28 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
             val readField = readDataSchema(writeAt)
             if (areNamesEquiv(clippedGroups, readAt, readField.name, isSchemaCaseSensitive)) {
               val origCol = table.getColumn(readAt)
-              val col: ColumnVector = if (typeCastingNeeded) {
-                ColumnCastUtil.ifTrueThenDeepConvertTypeAtoTypeB(origCol, readField.dataType,
-                  (dt, cv) => cv.getType.isDecimalType &&
-                      !GpuColumnVector.getNonNestedRapidsType(dt).equals(cv.getType()),
-                  (dt, cv) =>
-                    cv.castTo(DecimalUtil.createCudfDecimal(dt.asInstanceOf[DecimalType])))
+
+              var col: ColumnVector = origCol
+              if (!(decimalCastingNeeded || unsignedCastingNeeded)) {
+                col = origCol.incRefCount()
               } else {
-                origCol.incRefCount()
+                if (decimalCastingNeeded) {
+                  col = ColumnCastUtil.ifTrueThenDeepConvertTypeAtoTypeB(
+                    col, readField.dataType,
+                    (dt, cv) => cv.getType.isDecimalType &&
+                        !GpuColumnVector.getNonNestedRapidsType(dt).equals(cv.getType()),
+                    (dt, cv) =>
+                      cv.castTo(DecimalUtil.createCudfDecimal(dt.asInstanceOf[DecimalType])))
+                }
+                if (unsignedCastingNeeded) {
+                  col = ColumnCastUtil.ifTrueThenDeepConvertTypeAtoTypeB(
```

Comment:
```suggestion
                  col = ColumnCastUtil.ifTrueThenDeepConvertTypeAtoTypeB( // Add this for code change requirement.
```
The input `col` is going to be leaked.